### PR TITLE
Users/alsehr/bicep vwan

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Added tested VWAN bicep module
- *NOTE:* The removal needs custom logic to function. The pipeline will remain red for now and we need to manualy clean up. 

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.Network/virtualWans/deploy.bicep [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    builtInRoleNames: builtInRoleNames
+    wanName: wanName
+  }
+  dependsOn: [
+    virtualWan
+  ]
+}]
+
+output wanName string = wanName
+output wanNameResourceId string = virtualWan.id
```
Comment (MariusStorhaug) [inline arm/Microsoft.Network/virtualWans/deploy.bicep:195]: Should this not be virtualWanName, virtualWanResourceId and vWanResourceGroupName?
Also, should not virtualWanName string = virtualWan.name?

File: arm/Microsoft.Network/virtualWans/readme.md [hunk 2]
```diff
@@ -34,6 +35,7 @@ This template deploys Virtual Wan
 | `roleAssignments` | array | [] | Complex structure, see below. | Optional. Array of role assignment objects that contain the 'roleDefinitionIdOrName' and 'principalId' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or it's fully qualified ID in the following format: '/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11'
 | `tags` | object | {} | Complex structure, see below. | Optional. Tags of the Virtual Wan resource.
 | `cuaId` | string | "" | | Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered.
+| `lockForDeletion` | bool | False |  | Optional. Switch to lock storage from deletion. |
```
Comment (MariusStorhaug) [inline arm/Microsoft.Network/virtualWans/readme.md:38]: Should not reference storage, but virtualWan or so

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]
Review the comments.

File: arm/Microsoft.Network/virtualWans/deploy.bicep (hunk 1 above)
Comment (MrMCake) [reply arm/Microsoft.Network/virtualWans/deploy.bicep]: Agreed. I just left it as it is.
Regarding the reference of the name either seems fine. Updated it now

File: arm/Microsoft.Network/virtualWans/readme.md (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.Network/virtualWans/readme.md]: fixed

**Review by MariusStorhaug — APPROVED** [review]
Approved, as my comments are solved.

**Pull request merged**